pkgs/build-support/node/build-npm-package/hooks/npm-install-hook.sh

```diff
@@ -0,0 +1,30 @@
+npmInstallHook() {
+    echo "Executing npmInstallHook"
+
+    runHook preInstall
+
+    # `npm pack` writes to cache
+    npm config delete cache
+
+    local packageOut="$out/lib/node_modules/$pname"
+
+    while IFS= read -r file; do
+        local dest="$packageOut/$(dirname "$file")"
+        mkdir -p "$dest"
+        cp "$file" "$dest"
+    done < <(@jq@ --raw-output '.[0] | .files | map(.path) | join("\n")' <<< "$(npm pack --json --dry-run)")
+
+    while IFS=" " read -ra bin; do
+        mkdir -p "$out/bin"
+        makeWrapper @hostNode@ "$out/bin/${bin[0]}" --add-flags "$packageOut/${bin[1]}"
```

Comment:
We should definitely have something along the lines of `npmWrapperArgs` so that other hooks can append to that instead of having to double wrap.